lib/cocoapods/installer/target_installer/pod_target_installer.rb

```diff
@@ -66,9 +67,13 @@ def add_files_to_build_phases
           native_target.add_file_references(header_file_refs) do |build_file|
             # Set added headers as public if needed
             if target.requires_frameworks?
+              build_file.settings ||= {}
               if public_headers.include?(build_file.file_ref.real_path)
-                build_file.settings ||= {}
                 build_file.settings['ATTRIBUTES'] = ['Public']
+              elsif private_headers.include?(build_file.file_ref.real_path)
+                build_file.settings['ATTRIBUTES'] = ['Private']
+              else
+                build_file.settings['ATTRIBUTES'] = ['Project']
```

Comment:
There can be headers, which are neither public nor private, if you include them in the `source_files` and explicitly define `public_headers` and `private_headers`. Is there any reason to make that more explicit in the podspec? This would allow further conflicting definitions and make the whole logic to determine the visibility of a header more complex and harder to understand.
At the moment, the rules are pretty simple:
* If there are no explicit `public_headers`, all headers are assumed to be public.
* `private_headers` have higher precedence then `public_headers`.

We could add:
* `project_headers` have higher precedence then `private_headers`.